### ISSUE TITLE
npm-groovy-lint not finding Java installation

### ISSUE THREAD

**Issue by SohamJ** [nvuillam/npm-groovy-lint#101]
npm-groovy-lint is not able to find my local Java installation. 

The module version is as of this date (9/4) (couldn't get the exact version due to the error shown below) -

```bash
npm-groovy-lint --version
Java jre or jdk between 8 and undefined is required 
Installing Java jre 8 in /Users/<user>/.java-caller...
Unexpected error: request to https://api.adoptopenjdk.net/v2/info/releases/openjdk8?type=jre&openjdk_impl=hotspot&release=latest&os=mac&arch=x64& failed, reason: self signed certificate in certificate chain
FetchError: request to https://api.adoptopenjdk.net/v2/info/releases/openjdk8?type=jre&openjdk_impl=hotspot&release=latest&os=mac&arch=x64& failed, reason: self signed certificate in certificate chain
    at ClientRequest.<anonymous> (/usr/local/lib/node_modules/npm-groovy-lint/node_modules/node-fetch/lib/index.js:1455:11)
    at ClientRequest.emit (events.js:314:20)
    at TLSSocket.socketErrorListener (_http_client.js:469:9)
    at TLSSocket.emit (events.js:314:20)
    at emitErrorNT (internal/streams/destroy.js:100:8)
    at emitErrorCloseNT (internal/streams/destroy.js:68:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
```

Tried setting the java installation directly too - 
```bash
npm-groovy-lint --format --verbose --javaexecutable /usr/bin/java

# Same error as above
```
* OS - MacOS Mojave 10.15.13
* Java -

```bash
which java
/usr/bin/java

java -version 
openjdk version "1.8.0_265"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_265-b01)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.265-b01, mixed mode)
```

* Node -
```bash
node -v
v14.9.0
```

Please let me know if you need more information or If I missed anything. My groovylintrc is pretty minimal -

```json
{
  "extends": "all",
  "rules": {
    "formatting.Indentation": {
      "severity": "error",
      "spacesPerIndentLevel": 2
    }
  }
}
```

**Comment by nvuillam** [comment]
It seems there are 2 errors: 
- Unable to correctly detect Java 8 ( solved with PR #102 and new patch v7.5.5)
- When calling adoptopenjdk.com, SSL issue with your local or proxy Self-signed certificate

Once the first issue is solved, you shouldn't see the second one, please can you confirm that it's ok with v7.5.5, or reopen the issue if you still have an error ?

ps: Kudos for your bug description,  it's very clear and exhaustive :)

**Comment by SohamJ** [comment]
Thank you for the quick update @nvuillam! 

I updated to 7.6.1 today (I see its similar to 7.5.5 with the additions being related to Github actions, so went ahead with 7.6.1) -

It is now able to get the Java version to proceed further - 

```groovy
npm-groovy-lint -v    
GroovyLint: Successfully processed CodeNarc: 
CodeNarc version 1.6.1

npm-groovy-lint version 7.6.1

Embeds:
CodeNarc version 1.6.1
- Groovy version 3.0.5 (superlite)
```

On the second issue with self-signed cert, I still see it when I run the format command. The format command itself runs successfully, followed by the error. I confirmed that this is happening due to the insights. Enabling our proxy fixes the issue - thought I'd comment here in case someone else has the same question since the stacktrace didn't indicate the root cause.

```groovy
npm-groovy-lint --verbose --format

Unexpected error: self signed certificate in certificate chain
Error: self signed certificate in certificate chain
    at TLSSocket.onConnectSecure (_tls_wrap.js:1497:34)
    at TLSSocket.emit (events.js:314:20)
    at TLSSocket._finishInit (_tls_wrap.js:932:8)
    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:706:12)
```

Thanks for your help! I am looking forward to exploring the auto-formatting feature along with VSCode

**Comment by nvuillam** [comment]
I'm glad to know it works better on your configuration :)

I think the remaining issue is about the self-signed certificate, not related to npm-groovy-lint architecture, but if you don't want to see it, just send **--no-insight** option :)
